include/matrix_basic.hpp

```diff
@@ -293,41 +285,39 @@ Matrix Matrix::operator-(Matrix mat) {
 
     if ((row_length() == mat.row_length()) && (col_length() == mat.col_length())) {
         Matrix result;
-        std::vector<std::string> row;
+        std::vector<double> row;
 
         for (int i = 0; i < row_length(); i++) {
             for (int j = 0; j < col_length(); j++)
-                row.push_back(std::to_string(double_mat[i][j] - mat.double_mat[i][j]));
-            result.str_mat.push_back(row);
+                row.push_back(double_mat[i][j] - mat.double_mat[i][j]);
+            result.double_mat.push_back(row);
             row.clear();
         }
-        result.to_double();
+        result.to_string();
         return result;
     } else if ((row_length() == mat.row_length()) && (mat.col_length() == 1)) {
-        mat = vector_to_mat(mat);
         Matrix result;
-        std::vector<std::string> row;
+        std::vector<double> row;
 
         for (int i = 0; i < row_length(); i++) {
             for (int j = 0; j < col_length(); j++)
-                row.push_back(std::to_string(double_mat[i][j] - mat.double_mat[i][j]));
-            result.str_mat.push_back(row);
+                row.push_back(double_mat[i][j] - mat.double_mat[i][0]);
```

Comment:
Shouldn't it be: 

                LINE 304: row.push_back(double_mat[i][j] - mat.double_mat[i][j]);

Reply:
mat is a vector or n x 1 dimensional Matrix, for any arbitrary n. If I did mat.double_mat[I][j], where 0<=i<row_length and 0<=j<col_length, I will get a segmentation fault. Rest you can figure out how to tackle that problem.